src/storage/v2/inmemory/storage.cpp

```diff
@@ -663,6 +664,99 @@ Result<EdgeAccessor> InMemoryStorage::InMemoryAccessor::EdgeChangeType(EdgeAcces
   return EdgeAccessor(edge_ref, new_edge_type, from_vertex, to_vertex, storage_, &transaction_);
 }
 
+bool InMemoryStorage::InMemoryAccessor::CheckConstraintsOnPull() {
+  MG_ASSERT(is_transaction_active_, "The transaction is already terminated!");
+  MG_ASSERT(!transaction_.must_abort, "The transaction can't be committed!");
+
+  auto *mem_storage = static_cast<InMemoryStorage *>(storage_);
+
+  {
+    if (transaction_.constraint_verification_info.NeedsExistenceConstraintVerification()) {
+      const auto vertices_to_update =
+          transaction_.constraint_verification_info.GetVerticesForExistenceConstraintChecking();
+      for (auto const *vertex : vertices_to_update) {
+        // No need to take any locks here because we modified this vertex and no
+        // one else can touch it until we commit.
+        auto validation_result = storage_->constraints_.existence_constraints_->Validate(*vertex);
+        if (validation_result) {
+          Abort();
+          DMG_ASSERT(!commit_timestamp_.has_value());
+          return true;
+          // return StorageManipulationError{*validation_result};
+        }
+      }
+    }
+
+    // Result of validating the vertex against unqiue constraints. It has to be
+    // declared outside of the critical section scope because its value is
+    // tested for Abort call which has to be done out of the scope.
+    std::optional<ConstraintViolation> unique_constraint_violation;
+
+    // Save these so we can mark them used in the commit log.
+    uint64_t start_timestamp = transaction_.start_timestamp;
+
+    {
+      std::unique_lock<utils::SpinLock> engine_guard(storage_->engine_lock_);
+      auto *mem_unique_constraints =
+          static_cast<InMemoryUniqueConstraints *>(storage_->constraints_.unique_constraints_.get());
+      commit_timestamp_.emplace(
+          mem_storage->CommitTimestamp(std::nullopt));  // TODO ante std::nullopt = desired_commit_timestamp
```

Comment:
This is tricky part and should now be changed, before it happened at the commit time and was ok but not here